packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2111,7 +2124,14 @@ export default (superClass: Class<Parser>): Class<Parser> =>
                 // $FlowIgnore
                 node.optional = false;
               }
-              return this.finishCallExpression(node, state.optionalChainMember);
+              if (isOptionalCall) {
+                node.optional = true;
```

Comment:
When `state.optionalChainMember` is set, we can merge this branch with the preceding one.

Reply:
I couldn't understand the above code. I don't know the reason that if `state.optionalChainMember` is true, then `node.optional` would be false.
https://github.com/babel/babel/blob/b3990faed0378053db57621774151d23ba4bab6b/packages/babel-parser/src/plugins/typescript/index.js#L2127-L2130

You mean like this? 
```
if (state.optionalChainMember) {
  node.optional = isOptional? true: false;
}
```

Reply:
Yes. The `isOptional` should be assigned to `node.optional`.